Source/ASTextNode.mm

```diff
@@ -471,34 +458,40 @@ - (NSArray *)exclusionPaths
 
 #pragma mark - Drawing
 
+static NSString *ASTextNodeBackgroundColorDrawParameterKey = @"backgroundColor";
+static NSString *ASTextNodeContainerInsetsDrawParameterKey = @"containerInsets";
+static NSString *ASTextNodeRendererDrawParameterKey = @"renderer";
+
 - (NSObject *)drawParametersForAsyncLayer:(_ASDisplayLayer *)layer
 {
   ASDN::MutexLocker l(__instanceLock__);
-  
-  _drawParameter = {
-    .backgroundColor = self.backgroundColor,
-    .bounds = self.bounds
-  };
-  return nil;
+ 
+  NSMutableDictionary *drawParameters = [[NSMutableDictionary alloc] init];
+  if (self.backgroundColor != nil) {
+    drawParameters[ASTextNodeBackgroundColorDrawParameterKey] = self.backgroundColor;
+  }
+  if (!UIEdgeInsetsEqualToEdgeInsets(_textContainerInset, UIEdgeInsetsZero)) {
+    drawParameters[ASTextNodeContainerInsetsDrawParameterKey] = [NSValue valueWithUIEdgeInsets:_textContainerInset];
+  }
+  drawParameters[ASTextNodeRendererDrawParameterKey] = [self _rendererWithBoundsSlow:self.bounds];
```

Comment:
Getting this renderer is indeed slow so we shouldn't run it on the main thread – can we do one of these two things:

- Make a bigger change where `_rendererWithBoundsSlow:` becomes a class method
- Or make this a smaller change where we return `self` from this method and use it in `+drawRect:withParameters:(self)` and update this implementation later.